reframe/core/modules.py

```diff
@@ -939,3 +941,111 @@ def emit_load_instr(self, module):
 
     def emit_unload_instr(self, module):
         return ''
+
+
+class SpackImpl(ModulesSystemImpl):
+    '''Base class for TMod Module system (Tcl).'''
+
+    def __init__(self):
+        # Try to figure out if we are indeed using the TCL version
+        try:
+            completed = osext.run_command('spack -V')
+        except OSError as e:
+            raise ConfigError(
+                'could not find a sane Spack installation') from e
+
+        self._version = completed.stdout.strip()
+
+        # self._version = version
+        # try:
+        #     # Try the Python bindings now
+        #     completed = osext.run_command(self.modulecmd())
+        # except OSError as e:
+        #     raise ConfigError(
+        #         'could not get the Python bindings for TMod: ' % e) from e
+
+        # if re.search(r'Unknown shell type', completed.stderr):
+        #     raise ConfigError(
+        #         'Python is not supported by this TMod installation')
+
+    def name(self):
+        return 'spack'
+
+    def version(self):
+        return self._version
+
+    def modulecmd(self, *args):
+        return ' '.join(['spack', *args])
+
+    def _execute(self, cmd, *args):
+        modulecmd = self.modulecmd(cmd, *args)
+        completed = osext.run_command(modulecmd)
+        if re.search(r'Error', completed.stderr) is not None:
+            raise SpawnedProcessError(modulecmd,
+                                      completed.stdout,
+                                      completed.stderr,
+                                      completed.returncode)
+
+        return completed.stdout
+
+    def available_modules(self, substr):
+        output = self.execute('find', '--format', '{name}-{version}-{hash}',
+                              substr)
+        ret = []
+        for line in output.split('\n'):
+            if not line or line[-1] == ':':
+                # Ignore empty lines and path entries
+                continue
+
+            ret.append(Module(line))
```

Comment:
Are you should that this is going to be parsed correctly by `Module`? I think that `Module` expects module names as `name/version`.